data/unaligned_labeled_mask_dataset.py

```diff
@@ -76,7 +76,7 @@ def get_img(self,A_img_path,A_label_path,B_img_path=None,B_label_path=None,index
                 print("failed to read B domain image ", B_img_path, " or label", B_label_path)
                 return None
             
-            if len(self.B_label_paths) > 0: # B label is optional
+            if len(self.B_label_paths) > 0 and index_B is not None: # B label is optional
                 B_label_path = self.B_label_paths[index_B]
```

Comment:
Just remove that line and cancel all the other changes and it should be ok pour this fix :)